src/components/HistoryTables/MessageHistory/Detail/index.tsx

```diff
@@ -0,0 +1,202 @@
+import LoaderGlyph from '../../../LoaderGlyph'
+
+// add RelativeTime plugin to Day.js
+dayjs.extend(relativeTime.default)
+
+enum MessageState {
+  Loading,
+  // message is in the mempool
+  Pending,
+  // message is included in chain, but not yet executed (no gas cost, no receipt)
+  Confirmed,
+  // message has been fully executed - all variables should exist now
+  Executed,
+  // failed to get message
+  Error,
+  NotFound
+}
+
+export default function MessageDetail(props: MessageDetailProps) {
+  const { cid, speedUpHref, cancelHref, confirmations } = props
+  const time = useMemo(() => Date.now(), [])
+  const [seeMore, setSeeMore] = useState(false)
+  const { message, error, loading, pending } = useMessage(cid)
+
+  const { data: gasQuery } = useGasCostQuery({ variables: { cid } })
+  const { data: msgRcptQuery } = useMessageReceiptQuery({
+    variables: { cid }
+  })
+
+  const transactionFee = useMemo<string>(() => {
+    if (pending || !gasQuery?.gascost?.totalCost) return 'Pending...'
+    const cost = gasQuery?.gascost?.totalCost
+    return cost
+      ? `${makeFriendlyBalance(new FilecoinNumber(cost, 'attofil'))} FIL`
+      : ''
+  }, [gasQuery?.gascost?.totalCost, pending])
+
+  const gasUsed = useMemo(
+    () => gasQuery?.gascost?.gasUsed || 0,
+    [gasQuery?.gascost]
+  )
+
+  const { methodName, actorName } = useMethodName(message)
+
+  const execReturn = useMemo<ExecReturn | null>(() => {
+    if (!message) return null
+    const toExec = isAddrEqual(message?.to, 'f01')
+    const receiptExists = !!msgRcptQuery?.receipt?.return
+    const initTx = Number(message?.method) === 2
+    if (toExec && receiptExists && initTx) {
+      return getAddrFromReceipt(msgRcptQuery?.receipt?.return)
+    }
+    return null
+  }, [message, msgRcptQuery?.receipt])
+
+  const messageState = useMemo<MessageState>(() => {
+    if (pending) return MessageState.Pending
+    else if (loading) return MessageState.Loading
+    else if (!message && !pending && !error) return MessageState.NotFound
+    else if (!message && !pending && !!error) return MessageState.Error
+    else if (!!message && !gasUsed) return MessageState.Confirmed
+    else if (!!message && !!gasUsed) return MessageState.Executed
+    else return MessageState.Loading
+  }, [pending, message, error, gasUsed, loading])
+
+  return (
+    <>
+      <Head
+        title='Message Overview'
+        pending={pending}
+        speedUpHref={speedUpHref}
+        cancelHref={cancelHref}
+      />
+      <Lines>
+        {messageState === MessageState.Error && (
+          <DetailCaption
+            name='Message Overview'
+            captian='Error'
+            loading={false}
+            error={error}
+          />
+        )}
+        {messageState === MessageState.NotFound && (
+          <StandardBox>
+            <p>Message {cid} not found.</p>
+            <p>
+              Note - it may take 1-2 minutes for a recently confirmed message to
+              show up here.
```

Comment:
We could show a "our database is still syncing mainnet" thing here too, but i couldnt think of a good way to word / design this so i left it for now